sources/packages/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -0,0 +1,149 @@
+  transformToApplicationDto,
+  StudentApplicationAndCount,
+  ApplicationFormData,
+} from "./models/application.model";
+import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { UserGroups } from "../../auth/user-groups.enum";
+import { Application } from "../../database/entities";
+import {
+  FieldSortOrder,
+  DEFAULT_PAGE_NUMBER,
+  DEFAULT_PAGE_LIMIT,
+  transformToApplicationSummaryDTO,
+  getOfferingNameAndPeriod,
+} from "../../utilities";
+import { ApiNotFoundResponse, ApiOkResponse, ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("application")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-application`)
+export class ApplicationAESTController extends BaseController {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly offeringService: EducationProgramOfferingService,
+    private readonly locationService: InstitutionLocationService,
+    private readonly programService: EducationProgramService,
+  ) {
+    super();
+  }
+
+  /**
+   * API to fetch application details by applicationId.
+   * This API will be used by ministry users.
+   * @param applicationId
+   * @returns Application details
+   */
+  @Get(":applicationId")
+  @ApiOkResponse({ description: "Application details fetched." })
+  @ApiNotFoundResponse({ description: "Application not found." })
+  async getApplication(
+    @Param("applicationId") applicationId: number,
+  ): Promise<GetApplicationBaseDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+    const additionalFormData = {} as ApplicationFormData;
+    // Check wether the selected location is designated or not.
+    // If selected location is not designated, then make the
+    // selectedLocation null
+    if (application.data.selectedLocation) {
+      const designatedLocation =
+        await this.locationService.getDesignatedLocationById(
+          application.data.selectedLocation,
+        );
+      const selectedLocation = await this.locationService.getLocationById(
+        application.data.selectedLocation,
+      );
+      if (!designatedLocation) {
+        application.data.selectedLocation = null;
+      }
+      // Assign location name for readonly form
+      additionalFormData.selectedLocationName = selectedLocation.name;
+    }
+    // Check wether the program is approved or not.
+    // If selected program is not approved, then make the
+    // selectedLocation null
+    if (application.data.selectedProgram) {
+      const selectedProgram = await this.programService.getProgramById(
+        application.data.selectedProgram,
```

Comment:
selectedLocationName, selectedProgramName and selectedOfferingName can be part of applicationData as optional fields itself. Why is that declared as ApplicationFormData ?

Reply:
`applicationData`  is for the database layer. `selectedLocationName, selectedProgramName, and selectedOfferingName` will not be saved in database, it's only for loading initial data in read-only form

Reply:
The ApplicationData interface represents the _jsonB_ dynamic data saved into DB at the database layer.
If the UI layer needs something else that is not present in the DB it is not the responsibility of the ApplicationData to have it mapped.

Reply:
I get that point. Absolutely right. Still we can have single object interface created at UI side to represent the student application form and these fields can be an optional ones to that object.

Reply:
Ignore the point above. I got the gist.